File: PHP/Generator/IGenerator.php
<?php

namespace Generator\IGenerator;

use Generator\GeneratedFile\GeneratedFile;
use One\Ast\Types\Package;
use One\ITransformer\ITransformer;

interface IGenerator {
    function getLangName();
    
    function getExtension();
    
    function getTransforms();
    
    function generate($pkg);
}

File: CSharp/Generator/IGenerator.cs
using Generator;
using One.Ast;
using One;

namespace Generator
{
    public interface IGenerator {
        string getLangName();
        
        string getExtension();
        
        ITransformer[] getTransforms();
        
        void addPlugin(IGeneratorPlugin plugin);
        
        void addInclude(string include);
        
        string expr(IExpression expr);
        
        GeneratedFile[] generate(Package pkg);
    }
}